src/includes/traits/Plugin/AutoCacheUtils.php

```diff
@@ -170,5 +170,49 @@ public function autoCacheCheckXmlSitemap($sitemap, $is_nested_sitemap = false, $
 
         return true;
     }
+
+    /**
+     * Download XML Sitemap via cURL and store in a temp file.
+     *
+     * @since 16xxxx
+     *
+     * @param string $url to XML Sitemap that needs to be downloaded
+     *
+     * @throws \Exception If unable to write temporary file or download XML Sitemap
+     *
+     * @return string Path to downloaded XML Sitemap file
+     */
+    public function autoCacheGetSitemapViaCurl($url)
+    {
+        if (!($tmp_dir = $this->getTmpDir())) {
+            throw new \Exception(__('No writable tmp directory.', SLUG_TD));
+        }
+
+        $tmp_file = $tmp_dir.'/'.$this->addTmpSuffix($tmp_dir).'.xml';
+
+        if (!($fp = fopen($tmp_file, 'wb'))) {
+            throw new \Exception(sprintf(__('Unable to open tmp file for writing:  `%1$s`.', SLUG_TD), $tmp_file));
+        }
+
+        $ch = curl_init();
+
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_FILE, $fp);
+        curl_setopt($ch, CURLOPT_HEADER, 0);
+        curl_setopt($ch, CURLOPT_TIMEOUT, 10);
+        curl_setopt($ch, CURLOPT_FAILONERROR, true);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
```

Comment:
In this line, setting `CURLOPT_FOLLOWLOCATION` to `true` on an installation of PHP that has `safe_mode` on, or has an `open_basedir` is likely to result in an error due to a security precaution in PHP. That's bitten me in the past, so just wanted to let you know about this. See: https://github.com/WordPress/WordPress/blob/602b51a209e39462e45003378df2bf89d008aa72/wp-includes/class-wp-http-curl.php#L147

Or, you might consider converting this to `wp_remote_get()`, which will use cURL first and then automatically fallback on the `Http` extension if it exists. It also deals with quirks like the location redirect flag.

So something like...

```php
$response = wp_remote_get( ... );
if ($response && !is_wp_error($response)) {
    file_put_contents($tmp_file, $response['body']);
}
```